### PR TITLE
Update Model component + Introduction + add implementation of autotag to DevGuide

### PR DESCRIPTION
**Pull request by dorcastan**
To do:
- [x] Write implementation of `autotag` command, along with relevant diagrams.

Update next time:
- Improve sequence diagram
- Add activity diagram

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -116,22 +126,39 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
...
 
+The `Model` component:
+
+* Stores the data of Mark.
+* Does not depend on any of the other three components.
+
 .Structure of the Model Component
 image::ModelClassDiagram.png[]
 
 *API* : link:{repoURL}/src/main/java/seedu/address/model/Model.java[`Model.java`]
```
Comment (Na-Nazhou) [inline docs/DeveloperGuide.adoc]:
```suggestion
*API* : link:{repoURL}/src/main/java/seedu/mark/model/Model.java[`Model.java`]
```

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -16,6 +16,16 @@ endif::[]
 
 By: `Team CS2103T-T13-4`      Since: `Sept 2019`      Licence: `MIT`
 
+== Introduction
+
+Welcome to Mark. This developer guide aims to introduce potential developers to the structure and implementation of Mark,
```
Comment (kxyoke) [inline docs/DeveloperGuide.adoc]:
```suggestion
Welcome to Mark. This developer guide aims to introduce potential developers to the structure and implementation of Mark
```
Just a small suggestion (not sure if this is comma splicing)

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -116,22 +126,39 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
...
+`AutotagController`, and a `FolderStructure`.
+
+.Structure of the Annotation package
+image::AnnotationPackageClassDiagram.png[]
+
+The above diagram shows the structure of the Annotation package, which is accessed by `ModelManager` via `OfflineDocument`.
+An `OfflineDocument` consists of zero or more `Paragraph` objects. A `Paragraph` can be either a `PhantomParagraph`, which
+contains only an `AnnotationNote` and a `ParagraphIdentifier`, or a `TrueParagraph`, which contains `AnnotationNote`,
+`ParagraphIdentifier`, `Highlight`, and `ParagraphContent`. `AnnotationNote` and `Highlight` are optional fields
+whenever they are used, i.e. in `PhantomParagraph` and/or `TrueParagraph`.
```
Comment (kxyoke) [inline docs/DeveloperGuide.adoc]: Will it be better to place the brief preface before the diagram?(in CS2101 pov)

File: src/main/java/seedu/mark/model/ModelManager.java [hunk 4]
```diff
@@ -54,7 +54,7 @@ public ModelManager(ReadOnlyMark mark, ReadOnlyUserPrefs userPrefs) {
                 FXCollections.observableArrayList(
                         new OfflineDocument("example doc",
                                 OfflineDocument.OFFLINE_DOC_EXAMPLE).getCollection()), (
-            Paragraph p1, Paragraph p2) -> {
+                Paragraph p1, Paragraph p2) -> {
```
Comment (dorcastan) [inline src/main/java/seedu/mark/model/ModelManager.java]:
```suggestion
            Paragraph p1, Paragraph p2) -> {
```

**Pull request merged**